src/utils/rfqm_health_check.ts

```diff
@@ -76,3 +86,71 @@ export async function checkSqsQueueAsync(producer: Producer): Promise<HealthChec
     }
     return results;
 }
+
+/**
+ * Looks at the worker heartbeats and produces appropriate issues based on the age
+ * of the heartbeats and the worker balances.
+ *
+ * Heartbeat Age: Checks the most recent heartbeat and produces a FAILED issue if it is older than the failed
+ * threshold. For heartbeats other than the most recent, will only produce a DEGRADED issue. (i.e. the check only
+ * fails if ALL workers are stuck)
+ *
+ * Worker Balance: Like with the age check, this only produces a FAILED issue if all workers are below the failed
+ * balance. Individual worker balances produce a DEGRADED issue if they are below BALANCE_DEGRADED_THRESHOLD.
+ *
+ * Current date is an optional parameter for testing.
+ */
+export async function checkWorkerHeartbeatsAsync(
+    heartbeats: RfqmWorkerHeartbeatEntity[],
+    nowDate: Date = new Date(),
```

Comment:
ooh, when does this get evaluated? load time or run time? will this be set once only? or will it be recreated when the function executes?

Reply:
I believe that'll get evaluated when each time the function is called, but good to be certain

Reply:
checked:

```
function gimmeTime(nowDate = new Date()) {
    console.log(nowDate);
}

gimmeTime()
gimmeTime()
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

Reply:
awesome! this works differently in Python, where it's evaulated only once! thanks for checking.